### ISSUE TITLE
Staging Review finding: Modal opens on input

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#104798]
_Need help? Please review [how to read a Staging Review ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/anatomy-of-a-staging-review-issue-ticket). Tag `@platform-governance-team-members` on Slack if you need further assistance._

## Product Information
**Team:** Ask VA Integration
**Product:** Ask VA
**Feature:** Ask Va

## Findings details
**VA.gov Experience Standard - issue:** [User experiences predictable interactions with components and patterns.](https://design.va.gov/about/experience-standards/#user-can-identify-an-element-and-its-state)
**VA.gov Experience Standard - category:** [Usability](https://design.va.gov/about/experience-standards/#category-usability)
**Launch-blocking:** Yes
**Design System review:** No
**Collab Cycle Reviewer:** @briandeconinck (Accessibility), @allison0034 (Design)

### Description

On [/ask-va/category-topic-1](https://staging.va.gov/contact-us/ask-va/category-topic-1), if you select certain options in the category `va-select` element, a modal opens informing you that you need to sign in to ask this question.

Opening a modal constitutes a change of context that's not allowed under [WCAG success criterion 3.2.2](https://www.w3.org/WAI/WCAG20/Understanding/on-input). It's an unexpected change with no indication to the user that it may happen, and may occur when the user is still trying to understand their options.

For sighted users who have opened the `va-select` dropdown, there's some visual cues you can lean on to understand what's happening. But for screen reader users who don't see these visual cues, this is particularly disruptive.

When a user tabs to a `select`, you can open it to see the dropdown or you can use the arrow keys to move through the options without opening the dropdown at all. Many screen reader users choose to use the arrow keys without opening the dropdown, because it's fewer keystrokes and a blind user doesn't really benefit from opening something they can't see. But arrowing through the options also changes the value of the dropdown, which triggers the modal. (See video for a demonstration.)

Similar behavior also occurs on:
- [/ask-va/who-is-your-question-about](https://staging.va.gov/contact-us/ask-va/who-is-your-question-about)

<details><summary>Link, screenshot or steps to recreate</summary>

https://github.com/user-attachments/assets/f583326a-6c14-4f0d-b9f2-8ebb880e5e45

</details>

## Recommended action

I do not believe a modal is appropriate for these situations, but even if you keep a modal it is definitely not something that should open on input.

My best recommendation is to allow users to make their selection, click Continue, and conditionally display a new page with the message that a sign in is required.

## References

- **Accessibility Defect Severity:** [1: Critical. Must be fixed before launch.](https://depo-platform-documentation.scrollhelp.site/developer-docs/accessibility-defect-severity-rubric#Accessibilitydefectseverityrubric-a11y-defect-1)
- **WCAG Success Criteria:** [WCAG 3.2.2 On Input - Level A](https://www.w3.org/TR/WCAG21/#on-input)
- **Modality:** Screen reader
- **Design System Component:** N/A
- **Design System Pattern or Template:** N/A
- **Design System Foundation:** N/A
- **Content Style Guide:** N/A
- **Context:** N/A

---

### Next Steps for the VFS Team

_How did this touchpoint go? [Give feedback on the Collaboration Cycle](https://dj540s05.optimalworkshop.com/questions/f54355271969f160ef41182ba9ceb61c) at any time._
- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members` with any questions or to get help validating the issue.
- [ ] **Ticket ownership:** If you believe that this issue is out of scope, not your team's responsibility, or a Design System issue, comment and tag `@platform-governance-team-members` on your team channel in Slack to provide an explanation and who you believe is responsible. The Governance team will follow up.
- [x] **Close this ticket** when the issue has been resolved or validated by your Product Owner.

**Comment by arielma-tw** [comment]
Issue has been addressed [here](https://github.com/department-of-veterans-affairs/ask-va/issues/1622)
